storage/src/state/ledger.rs

```diff
@@ -100,6 +100,20 @@ impl<N: Network> LedgerState<N> {
         Self::open_writer_with_increment::<S, P>(path, 10_000)
     }
 
+    fn new<S: Storage>(storage: S, is_read_only: bool) -> Result<LedgerState<N>> {
```

Comment:
Can we leave this undeduped?

Reply:
Sure; let me know if the rest is good to go, and I can merge this myself after dropping this one commit.

Reply:
Yes good to go